runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -171,4 +186,72 @@ public void onMenuEntryAdded(MenuEntryAdded menuEntryAdded)
 			menuEntries[menuEntries.length - 1].setTarget("*" + menuEntries[menuEntries.length - 1].getTarget());
 		}
 	}
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired event)
+	{
+		if (event.getScriptId() != ScriptID.HP_HUD_UPDATE)
+		{
+			return;
+		}
+
+		clientThread.invoke(this::updateBossHealthBarText);
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getKey().equals("hitpointsDisplayStyle"))
+		{
+			return;
+		}
+
+		clientThread.invoke(this::updateBossHealthBarText);
+	}
+
+	/**
+	 *
+	 * Update the in-game boss health bar overlay text to what the user's config specifies.
+	 * This health bar is used in CoX, ToA, Gauntlet, quest bosses, etc. It is not used in ToB, which has its own.
+	 */
+	private void updateBossHealthBarText()
+	{
+		Widget widget = client.getWidget(WidgetInfo.HEALTH_OVERLAY_BAR_TEXT);
+		if (widget == null)
+		{
+			return;
+		}
+
+		final int currHp = client.getVarbitValue(Varbits.BOSS_HEALTH_CURRENT);
+		final int maxHp = client.getVarbitValue(Varbits.BOSS_HEALTH_MAXIMUM);
+
+		final String text;
+		switch (config.hitpointsDisplayStyle())
+		{
+			case HITPOINTS:
+				text = getHpText(currHp, maxHp);
+				break;
+			case PERCENTAGE:
+				text = getPercentText(currHp, maxHp);
+				break;
+			case BOTH:
+				text = getHpText(currHp, maxHp) + " (" + getPercentText(currHp, maxHp) + ")";
```

Comment:
`getHpText(currHp, maxHp)` is the current value so you can just delete this and replace it with `widget.getText()`

Reply:
See comment above